Google-Summer-of-Code-2023.md

```diff
@@ -0,0 +1,280 @@
+
+**Q: I do not have any experience in skill XYZ. Is some certification required?**
+
+A: Try to work on good first issues and take courses online. In the field of software development, it is common to develop experience and expertise as you take up and complete projects successfully. We do not require any formal certification of particular skills.
+
+**Q: Is it okay if I only focus on the frontend or backend?**
+
+A: This probably depends on the project(s) you wish to apply for. However, note that the ability to be effective in both the frontend and backend will open up more opportunities for you, since projects often touch multiple layers of the stack.
+
+**Q: The [Google GSoC FAQ](https://developers.google.com/open-source/gsoc/faq#can_someone_already_participating_in_open_source_be_a_gsoc_contributor) mentions that the program is only for new contributors. I have already contributed to Oppia and I have write access. Can I still participate?**
+
+A: The GSoC program is open to new and beginner contributors to open source and to students. Some of our contributors with write access are still beginner contributors, whereas some of our other contributors with write access will not qualify because they are experienced contributors. If you have only recently received write access, or have been contributing to Oppia for less than a year, you are probably still a beginner contributor. If the previous sentence does not apply to you, and you want to know which group you fall into, please contact **@srijanreddy98**, **@U8NWXD** or **@aasiffaizal** for a decision.
+
+**Q: I'd love to contribute to open source, but I'm not sure I have enough time during the summer to do a GSoC project. Can I still help out?**
+
+A: Yes, GSoC might not be the best choice if you don't have enough time during the summer. However, you can start contributing to Oppia by following the instruction in the contributors' guide carefully ([Oppia Web](https://github.com/oppia/oppia/wiki/Contributing-code-to-Oppia#setting-things-up), [Oppia Android](https://github.com/oppia/oppia-android/wiki/Contributing-to-Oppia-android)).
+
+**Q: Why doesn't Oppia provide difficulty levels for projects?**
+
+A: Difficulty can vary from person to person depending on the skills required which is why we list all the prerequisite skills of each project.
+ 
+## Dates and Deadlines
+
+Noteworthy dates for 2023 ([Full Timeline](https://developers.google.com/open-source/gsoc/timeline)):
+
+- **Jan 23 - Feb 7**: Mentoring organizations apply
+- **Feb 22**: Mentoring organizations are announced
+- **Mar 20 - Apr 4**: GSoC contributor application period
+- **May 4**: Accepted GSoC contributors are announced
+- **May 4 - May 28**: Community bonding period
+- **May 29 - Jul 14**: Milestone 1 work period for GSoC contributors
+- **Jul 15 - Jul 16**: Internal evaluation for M1
+- **Jul 17 - Jul 26**: Optional buffer period for M1 revisions
```

Comment:
"Optional" makes it sound like contributors can leave this buffer out of their proposal timelines, but I don't think we're allowing that. I think we can remove "optional" and just call it a "buffer period"

Reply:
Yeah, maybe we explicitly say that they should only be planning revisions for that period

Reply:
I do still disagree with the current text. "Optional" makes it sound like students can opt to skip the buffer period and, for example, plan to start M2 on Jul 17. That's misleading. Instead, the buffer period is a required part of our project timeline for this year.

I can see why one might want to include the word "optional" to convey that if no issues are raised during the evaluation, the student can move ahead. However, this doesn't really make the buffer period "optional" for the student since its dependent on a decision by the evaluator. Further, this property of the student being able to work ahead of schedule is true throughout the project, so it's inconsistent to only call it out for the buffer periods.

I think we should drop the word "optional" since the buffer period is a required part of our timeline. I'm fine with leaving this timeline high-level and not delving into what students do during the buffer period, but I don't think we should include "optional" since that implies something incorrect

Reply:
One other suggestion (y'all might want to discuss): you could just collapse these three lines and say:

- "May 29 - Jul 26": Milestone 1 work period (or "standard Milestone 1 work period")
- "... - ..." : ditto (for M2)

and leave all the other stuff out of it, since it can be (and has been) elaborated on in the proposal template.

Reply:
I think let's go with the collapsed timeline. I have collapsed the three lines to "Standard Milestone X work period for GSoC contributors"